src/vendor/components/navigation/index.jsx

```diff
@@ -0,0 +1,132 @@
+// eslint-disable react/prop-types
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import isEqual from 'lodash/isEqual';
+import { frum, zipObject } from '../../queryOps';
+import { Object2URL, URL2Object } from '../../convert';
+
+function withNavigation(config) {
+  // https://reactjs.org/docs/higher-order-components.html
+  //
+  // Deals with synchronizing url parameters with this.state, and
+  // selector widget states, and url history
+  //
+  // Expects a `config` that is a list of configurations for various
+  // selection widgets. Right now there is only Picker.
+  //
+  // Adds properties to `props`:
+  // * `navigation` - that contains a Navigation component
+  //   to include on `render()`
+  // * properties with names frum(config).select("id")
+
+  return WrappedComponent => {
+    class Output extends React.Component {
+      constructor(props) {
+        super(props);
+        const { location } = props;
+        const params = URL2Object(location.search);
+
+        this.params = params;
+
+        // SET PARAMETERS TO DEFAULT VALUES, OR URL PARAMETER
+        this.state = frum(config)
+          .map(({ id, defaultValue }) => [params[id] || defaultValue, id])
+          .args()
+          .fromPairs();
+
+        this.updateHistory({});
+      }
+
+      onPathChange(event) {
+        const { name, value } = event.target;
+        const change = zipObject([name], [value]);
+
+        this.setState(change);
+
+        this.updateHistory(change);
+      }
+
+      updateHistory(change) {
+        const { history, location } = this.props;
+        const newState = { ...this.state, ...change };
+        const oldState = URL2Object(location.search);
+
+        if (isEqual(newState, oldState)) return;
+
+        const query = Object2URL(newState);
+
+        history.push(`${location.pathname}?${query}`);
+      }
+
+      navComponents() {
+        const { classes } = this.props;
+        const params = this.state;
+        const self = this;
+
+        return (
+          <div className={classes.root}>
+            {config.map(c => {
+              const { type, id, label, options } = c;
+
+              return React.createElement(type, {
+                key: id,
+                id,
+                label,
+                handleChange: (...args) => self.onPathChange(...args),
```

Comment:
You should be able to use `this` instead of `self` by changing `navComponents() {` to be `navComponents = () => {`.

Reply:
I was attempting `handleChange: this.onPathChange`, but function is not bound to `this` at call time.

I changed 

    onPathChange(event) {

to

    onPathChange = event => {

which seemed to solve the binding problem